lib/tests/streamlit/runtime/runtime_test.py

```diff
@@ -0,0 +1,118 @@
+        self.assertEqual(RuntimeState.INITIAL, self.runtime.state)
+
+        await self.start_runtime_loop()
+        self.assertEqual(RuntimeState.NO_SESSIONS_CONNECTED, self.runtime.state)
+
+        self.runtime.stop()
+        await asyncio.sleep(0)  # Wait a tick for the stop to be acknowledged
+        self.assertEqual(RuntimeState.STOPPING, self.runtime.state)
+
+        await self.runtime.stopped
+        self.assertEqual(RuntimeState.STOPPED, self.runtime.state)
+
+    async def test_create_session(self):
+        """We can create and remove a single session."""
+        await self.start_runtime_loop()
+
+        session_id = self.runtime.create_session(
+            client=MagicMock(spec=SessionClient), user_info=MagicMock()
+        )
+        self.assertEqual(
+            RuntimeState.ONE_OR_MORE_SESSIONS_CONNECTED, self.runtime.state
+        )
+
+        self.runtime.close_session(session_id)
+        self.assertEqual(RuntimeState.NO_SESSIONS_CONNECTED, self.runtime.state)
+
+    async def test_multiple_sessions(self):
+        """Multiple sessions can be connected."""
+        await self.start_runtime_loop()
+
+        session_ids = []
+        for ii in range(3):
+            session_id = self.runtime.create_session(
+                client=MagicMock(spec=SessionClient),
+                user_info=MagicMock(),
+            )
+
+            self.assertEqual(
+                RuntimeState.ONE_OR_MORE_SESSIONS_CONNECTED, self.runtime.state
+            )
+            session_ids.append(session_id)
+
+        for ii in range(len(session_ids)):
+            self.runtime.close_session(session_ids[ii])
+            expected_state = (
+                RuntimeState.NO_SESSIONS_CONNECTED
+                if ii == len(session_ids) - 1
+                else RuntimeState.ONE_OR_MORE_SESSIONS_CONNECTED
+            )
+            self.assertEqual(expected_state, self.runtime.state)
+
+        self.assertEqual(RuntimeState.NO_SESSIONS_CONNECTED, self.runtime.state)
+
+    async def test_close_invalid_session(self):
+        """Closing a session that doesn't exist is a no-op: no error raised."""
+        await self.start_runtime_loop()
+
+        # Close a session that never existed
+        self.runtime.close_session("no_such_session")
+
+        # Close a valid session twice
+        session_id = self.runtime.create_session(
+            client=MagicMock(spec=SessionClient), user_info=MagicMock()
+        )
+        self.runtime.close_session(session_id)
+        self.runtime.close_session(session_id)
+
+    async def test_is_active_session(self):
+        """`is_active_session` should work as expected."""
+        await self.start_runtime_loop()
+        session_id = self.runtime.create_session(
+            client=MagicMock(spec=SessionClient), user_info=MagicMock()
+        )
+        self.assertTrue(self.runtime.is_active_session(session_id))
+        self.assertFalse(self.runtime.is_active_session("not_a_session_id"))
```

Comment:
Would also probably be worthwhile to close the active session and verify that `is_active_session` is False